### PR TITLE
Timeline grid

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Initial timeline grid structure
Smart snapping and resizing included

### CODE REVIEW

**Comment by geenen124** [comment]
What are the dragging numbers?

File: src/main/java/gui/SnappingPane.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package gui;
+
+import javafx.scene.layout.Pane;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Created by Bart.
```
Comment (Markvanderuit) [inline src/main/java/gui/SnappingPane.java:8]: Add an explanation of what this is used for?

File: src/main/java/gui/TimetableBlock.java [hunk 2]
```diff
@@ -37,91 +43,120 @@
...
-            }
-        });
+                    double yCoordinate = e.getSceneY() - thisBlock.getHeight() / 2;
+                    SnappingPane myPane = pane.getMyPane(e.getSceneX(), yCoordinate);
+                    if (myPane != null) {
+                        pane.getGrid().getChildren().remove(thisBlock);
+                        pane.getGrid().add(thisBlock, myPane.getColumn(), myPane.getRow());
+                    }
+                } else if (draggingNumber == 1 || draggingNumber == 3) {
+                    int numCounts = (int) Math.round(dummyPane.getHeight() / pane.getCountHeight());
```
Comment (geenen124) [inline src/main/java/gui/TimetableBlock.java:132]: dragging number enum would be useful here?

**Comment by mennoo1996** [comment]
Good work!

**Pull request merged**